mergin/report.py

```diff
@@ -0,0 +1,237 @@
+        project (str): full project name (<namespace/project>).
+        since (str): starting project version tag, for example 'v3'.
+        to (str): ending project version tag, for example 'v6'.
+        out_dir (str): output directory to save report.csv in, defaults to temp dir
+    """
+    mp = MerginProject(directory)
+    mp.log.info(f"--- Creating changesets report for {project} from {since} to {to} versions ----")
+    versions_map = {v["name"]: v for v in mc.project_versions(project, since, to)["versions"]}
+    headers = ["file", "table", "author", "timestamp", "version", "quantity_type", "quantity"]
+    records = []
+    info = mc.project_info(project, since=since)
+    # filter only .gpkg files
+    files = [f for f in info["files"] if mp.is_versioned_file(f["path"])]
+    for f in files:
+        mp.log.debug(f"analyzing {f['path']} ...")
+        diff_file = os.path.join(mp.meta_dir, f["path"] + ".diff")
+        try:
+            if "history" not in f:
+                continue
+
+            # download diffs in desired range, in case versions are not within history,
+            # pass unknown, and it will be determined in download function
+            # it can be different for each file
+            since_ = since if since in f["history"] else None
+            to_ = to if to in f["history"] else None
+            if not (since_ and to_):
+                continue  # no change at all for particular file in desired range
+
+            mc.get_file_diff(directory, f["path"], diff_file, since_, to_, True)
+
+            # download full gpkg in "to" version to analyze its schema to determine which col is geometry
+            full_gpkg = os.path.join(mp.meta_dir, ".cache", f["path"])
+            if not os.path.exists(full_gpkg):
+                mc.download_file(directory, f["path"], full_gpkg, to)
+
+            # get gpkg schema
+            schema_file = full_gpkg + '-schema'  # geodiff writes schema into a file
+            if not os.path.exists(schema_file):
+                mp.geodiff.schema("sqlite", "", full_gpkg, schema_file)
+            with open(schema_file, 'r') as sf:
+                schema = json.load(sf).get("geodiff_schema")
+
+            # add records for every version (diff) and all tables within geopackage
+            for idx in range(int_version(since) + 1, int_version(to) + 1):
+                version = "v" + str(idx)
+                # skip version if there was no diff file
+                if version not in f['history']:
+                    continue
+                v_diff_file = os.path.join(mp.meta_dir, '.cache',
+                                           version + "-" + os.path.basename(f['history'][version]['diff']['path']))
+
+                version_data = versions_map[version]
+                cr = mp.geodiff.read_changeset(v_diff_file)
+                rep = ChangesetReport(cr, schema)
+                report = rep.report()
```

Comment:
I wonder if it makes sense to have this as an `ChangesetReport` object if we just create it, then call its method and never use it again - maybe just turn it into an ordinary function call?